tests/homework18/homework18.spec.js

```diff
@@ -0,0 +1,207 @@
+  await lastNameInput.blur();
+
+  await expect(lastNameErrorMessage).toHaveText('Last name required');
+});
+
+test('Invalid Last Name data - Wrong length', async ({ page }) => {
+  const lastNameInput = page.locator('input[name="lastName"]');
+  const lastNameErrorMessage = page.locator('div.invalid-feedback p');
+  await lastNameInput.fill('a');
+  await lastNameInput.focus();
+  await lastNameInput.blur();
+  await expect(lastNameErrorMessage).toHaveText('Last name has to be from 2 to 20 characters long');
+
+});
+
+test('Invalid Last Name data - Wrong characters', async ({ page }) => {
+  const lastNameInput = page.locator('input[name="lastName"]');
+  const lastNameErrorMessage = page.locator('div.invalid-feedback p');
+  await lastNameInput.fill('123');
+  await lastNameInput.focus();
+  await lastNameInput.blur();
+  await expect(lastNameErrorMessage).toHaveText('Last name is invalid');
+});
+
+test('Invalid Last Name data - Exceeds maximum length', async ({ page }) => {
+  const lastNameInput = page.locator('input[name="lastName"]');
+  const lastNameErrorMessage = page.locator('div.invalid-feedback p');
+  await lastNameInput.fill('A'.repeat(25)); 
+  await lastNameInput.focus();
+  await lastNameInput.blur();
+  await expect(lastNameErrorMessage).toHaveText('Last name has to be from 2 to 20 characters long');
+});
+
+//Email Field
+test('Empty Email field validation', async ({ page }) => {
+  const emailInput = page.locator('input[name="email"]');
+  const emailInputErrorMessage = page.locator('div.invalid-feedback p');
+  await emailInput.fill('');
+  await emailInput.focus();
+  await emailInput.blur();
+  await expect(emailInputErrorMessage).toHaveText('Email required');
+});
+
+test('Invalid Email data - Incorrect format', async ({ page }) => {
+  const emailInput = page.locator('input[name="email"]');
+  const emailInputErrorMessage = page.locator('div.invalid-feedback p');
+  await emailInput.fill('test.email');
+  await emailInput.focus();
+  await emailInput.blur();
+  await expect(emailInputErrorMessage).toHaveText('Email is incorrect');
+});
+//Password Field
+ test('Empty Password field validation', async ({ page }) => {
+  const passwordInput = page.locator('input[name="password"]');
+  const passwordError = page.locator('div.invalid-feedback p');
+  await passwordInput.fill('');
+  await passwordInput.focus();
+  await passwordInput.blur();
+  await expect(passwordError).toHaveText('Password required');
+  });
+
+  test('Invalid Password data - Incorrect format', async ({ page }) => {
+    const passwordInput = page.locator('input[name="password"]');
+    const passwordError = page.locator('div.invalid-feedback p');
+    await passwordInput.fill('test.email',{ timeout: 2000 });
+    await passwordInput.focus();
+    await passwordInput.blur();
+    await expect(passwordError).toHaveText('Password has to be from 8 to 15 characters long and contain at least one integer, one capital, and one small letter');  });
```

Comment:
check input border color

Reply:
await expect(nameErrorMessage).toHaveCSS('border-color', 'rgb(220, 53, 69)');

Reply:
why do you check border color only for name input? apply this check for other inputs too